### ISSUE TITLE
Only first item in IGNORE_VIEW_NAMES is ever processed

### ISSUE THREAD

**Issue by OmenApps** [CleitonDeLima/django-login-required-middleware#5]
Looks like the very first time through the for-loop, either the first view name matches, and we do nothing, or it doesn't match, and we redirect to login.

That means that any time we don't get a match on the first loop through, we automatically redirect to login regardless of whether the other views in IGNORE_VIEW_NAMES (or later on, IGNORED_PATHS) actually match our current path.

**Comment by CleitonDeLima** [comment]
Thanks for reporting a bug, I will fix it as soon as possible. :+1:

**Comment by OmenApps** [comment]
I'm not in a state to be writing code right now, but maybe something like this?

```
class LoginRequiredMiddleware(MiddlewareMixin):
    def process_request(self, request):
        assert hasattr(request, 'user'), (
            'The LoginRequiredMiddleware requires authentication middleware '
            'to be installed. Edit your MIDDLEWARE setting to insert before '
            "'django.contrib.auth.middleware.AuthenticationMiddleware'."
        )

        path = request.path
        if not request.user.is_authenticated:

            if request.is_ajax():
                raise PermissionDenied()

            try:
                resolver = resolve(path)
            except Resolver404:
                return redirect_to_login(path)

            views = ((name in resolver) for name in IGNORE_VIEW_NAMES)
            if not any(views):
                if not any(url.match(path) for url in IGNORE_PATHS):
                    return redirect_to_login(path)
```

**Comment by CleitonDeLima** [comment]
Resolved in PR #6 

I removed Resolver404, believe that this validation is irrelevant, considering that it is a 404 error.

Thanks again for reporting this :smile:
